tests/measurements/test_counts.py

```diff
@@ -170,9 +174,65 @@ def test_counts_shape_single_measurement_value(self):
         result = qml.counts(mv).process_samples(samples, wire_order=[0])
 
         assert len(result) == 2
-        assert set(result.keys()) == {"0", "1"}
-        assert result["0"] == np.count_nonzero(samples[:, 0] == 0)
-        assert result["1"] == np.count_nonzero(samples[:, 0] == 1)
+        assert set(result.keys()) == {0, 1}
+        assert result[0] == np.count_nonzero(samples[:, 0] == 0)
+        assert result[1] == np.count_nonzero(samples[:, 0] == 1)
```

Comment:
if we're not casting to `int` anymore, how are these test changes not failing?

Reply:
oops, that's keys not values. lgtm

Reply:
Talked about it offline. The casting was being done for the values, not the keys. Regardless, the change has been reverted.